src/StorageSync/StorageSync/Common/StorageSyncClientWrapper.cs

```diff
@@ -146,6 +153,24 @@ public string AfsAgentInstallerPath
             }
         }
 
+        /// <summary>
+        /// Gets the current application identifier.
+        /// </summary>
+        /// <value>The current application identifier.</value>
+        public Guid CurrentApplicationId
+        {
+            get
+            {
+                switch (AzureRmProfileProvider.Instance?.Profile?.DefaultContext?.Environment?.Name)
+                {
+                    case EnvironmentName.AzureUSGovernment:
```

Comment:
Ideally, this should be discoverable via API.  Failing that, this should be an extended property of the built-in environment, meaning that you can just get the value from the environment in the same way.

For a preview this is OK, but please file an issue to either add a discovery API, or add this value to the built-in environment extensions

Reply:
https://github.com/Azure/azure-powershell/issues/8652